### ISSUE TITLE
Add support for generic org.jooq.log.<logger>=threshold system properties

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15050]
jOOQ by default tries to class-load log4j, and if it doesn't find log4j, will default to `java.util.logging`. Both frameworks, (including most log4j "backends") allow for fine grained control over log output. But jOOQ users may not always be in control of said logging backends, so they will continue to request new flags for specific log messages.

Much rather than adding new flags (such as `-Dorg.jooq.no-logo` or `-Dorg.jooq.no-tips`), why not support more generic `-Dorg.jooq.log.<logger>` flags, which influence the `JooqLogger` globally, preventing logs before jOOQ sends them to the relevant backend. This would solve this problem once and for all, for all users, without having to discuss new flags again.

This will be implemented as system properties only, for now. Upon user request, other mechanisms can be designed as well, knowing that this is only an "escape hatch" for special cases. Most users should instead configure their loggers.

An example is here:
https://github.com/jOOQ/jOOQ/issues/11355#issuecomment-1537175092

**Comment by lukaseder** [comment]
Probably makes sense to allow for suppressing the `ERROR` level as well: https://github.com/jOOQ/jOOQ/issues/15051

**Comment by lukaseder** [comment]
This change will implement an internal refactoring of the `JooqLogger` contents. The previous `boolean` flags:

```java
private boolean                   supportsTrace   = true;
private boolean                   supportsDebug   = true;
private boolean                   supportsInfo    = true;
private boolean                   supportsWarn    = true;
```

... will be replaced by a single `Log.Level threshold` property. The flags have been put in place for (undocumented) reflective usage, so there is a chance that some users may already rely on this (undocumented) feature. For that reason, I will not backport this improvement despite the apparent utility here:
https://github.com/jOOQ/jOOQ/issues/11355#issuecomment-1537175092

Given that this was undocumented, I will not tag this change as a backwards incompatible change, nonetheless. The properties were `private`, and users did tamper with them at their own risk. The new feature, however, will be documented.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0

**Comment by lukaseder** [comment]
Hmm, the existing implementation reads the system property only at logger construction time. Since loggers are stored as `static` variables, that happens at classloading time only. It's not possible to change the logger state thereafter, if initialised with a `System` property. I guess it's worth improving that.
